internal/apiclient/pricing.go

```diff
@@ -29,8 +29,8 @@ type PriceQueryResult struct {
 	Result gjson.Result
 }
 
-func NewPricingAPIClient(cfg *config.Config) *PricingAPIClient {
-	currency := cfg.Currency
+func NewPricingAPIClient(cfg *config.RunContext) *PricingAPIClient {
```

Comment:
No biggy, but it seems like `ctx` would be a more descriptive variable name with the updated type. What do you think?
```suggestion
func NewPricingAPIClient(ctx *config.RunContext) *PricingAPIClient {
```

Reply:
good shout